libs/server/ACL/ACLParser.cs

```diff
@@ -191,6 +241,72 @@ public static void ApplyACLOpToUser(ref User user, string op)
             {
                 throw new ACLUnknownOperationException(op);
             }
+
+            // There's some fixup that has to be done when parsing a command
+            static bool TryParseCommandForAcl(string commandName, out RespCommand command)
```

Comment:
Ah, I feel almost exactly the opposite - I prefer (static) local methods when the method is only useful in one context, since it doesn't pollute autocomplete and clearly signals "this is a helper for this _single_ method".

But when in Rome and all, I'll move it out.

Reply:
Maybe we should start following this style in future. I've seen this in plenty of places, and your points make sense. @TedHartMS?

Reply:
> Ah, I feel almost exactly the opposite - I prefer (static) local methods when the method is only useful in one context, since it doesn't pollute autocomplete and clearly signals "this is a helper for this _single_ method".
> 
> But when in Rome and all, I'll move it out.

Makes sense, and I was expecting the pushback for the exact reasons you mentioned.
I wish there was a way to declare it as local outside the encapsulating method using a keyword or something ( .net people are you listening? 😄)
Maybe a cleaner way to use it will be to declare it at the top of the method.
I would not want to declare more than a couple in any method, @badrishc if you want to follow that style in other places.

Reply:
I'm a fan of local functions for the reasons @kevin-montrose states. Out of curiosity I just BDN'd a trivial local function vs inlined member and the local function showed a couple % faster.

In this case though I can see @vazois' point that it is kind of hidden due to the size of the calling function. But I still go with the "I only use it here so I only want to see it here" principle.

Reply:
Okay then! Let's stick with what Kevin has, for this case.